cfme/tests/containers/test_container_provider_crud.py

```diff
@@ -0,0 +1,71 @@
+import fauxfactory
+import pytest
+
+from cfme.common.provider_views import ContainerProvidersView
+from cfme.containers.provider import ContainersProvider
+from cfme.fixtures.provider import setup_or_skip
+from cfme.utils.appliance.implementations.ui import navigate_to
+
+
+pytestmark = [
+    pytest.mark.tier(1),
+    pytest.mark.provider([ContainersProvider], scope='function')
+]
+
+
+@pytest.mark.tier(3)
+def test_container_provider_crud(request, appliance, provider):
+
+    """
+    Polarion:
+        assignee: juwatts
+        caseimportance: medium
+        casecomponent: Containers
+        initialEstimate: 1/6h
+    """
+
+    if provider.exists:
+        provider.delete()
+
+    setup_or_skip(request, provider)
+
+    request.addfinalizer(lambda: provider.delete_if_exists(cancel=False))
```

Comment:
IIRC delete_if_exists default behavior is cancel=False, so you don't need a lambda here and can just pass the function.

Reply:
`delete_if_exists` passes *args and **kwargs to `delete`. Default behavior of delete is `cancel=True`. I think we need to change that to `False`. I will make and enchancement